proposals/2/README.md

```diff
@@ -0,0 +1,219 @@
+```
+
+And for it to just work and generically add those to the right namespace.
+
+## Background
+
+The current apps model lets you add/remove charts fine - but there are a number of limitations:
+
+### Only works in 1 namespace
+
+Currently apps can only work inside the dev namespace - we can’t support things like knative, gloo, cert-manager, nginx, velero which tend to be installed in separate namespaces.
+
+### Does not handle boot phases
+
+This is not surprising as we created the App extension model before boot. But right now boot has a number of distinct phases with pipeline steps between them where we may want to add apps:
+
+* pre/post ingress setup (in the case of nip.io / nginx-controller)
+* pre/post vault setup
+* pre/post vault population of parameters
+* pre/post setup of certmanager/external dns
+
+There is currently no way for an apps to specify a namespace + specify which phase its to be installed in & have (say) the domain / certs / secrets injected easily.
+
+So rather than adding steps at the right point in a list of steps in the jenkins-x.yml it would be nice to have a more declarative YAML file to describe which apps are to be enabled.
+
+E.g. we need a new YAML file with list of charts to be installed along with metadata for which phase to install them. We also want to invoke a selection of those appss to be installed at the different phases of the boot pipeline.
+
+### Limitations of the current deploy model
+
+We have some issues with the current way we deploy apps in Staging / Production:
+
+#### Composite charts hide the individual versions
+
+Right now we deploy all charts in Staging/Production as a single chart. That means if you use `helm list` we don't show any actual versions of the dependent charts - its just 1 chart (with no version).
+
+#### Removing Apps on Dev doesn't work
+
+If you remove a chart from `env/requirements.yaml`  in the Dev repository the pipeline does not remove the chart. e.g. if you add `lighthouse` and remove `prow` you need to manually delete the prow resources yourself.
+
+
+### Limitation of boot
+
+We currently manually configure in the boot pipeline the install of multiple systems usually in separate namespaces:
+
+* velero (if enabled)
+* nginx
+* externaldns
+* cert-manager
+* acme certs chart
+
+Ideally those should be modular and optional. e.g it should be really trivial to disable nginx if folks are using, say, istio - without having to hack a pipeline yaml.
+
+We'd also like to make it easy to add a number of other systems in order in separate namespaces independently of the `Staging / Production` environment namespaces:
+
+* knative https://github.com/jenkins-x/jx/issues/6331
+* gloo
+* istio
+* linkerd https://github.com/jenkins-x/jx/issues/6330
+* flagger
+
+so it'd be nice to have a simple app model where anyone can add any systems/charts to any namespace at any point in the flow before we setup the dev/staging/production environment in the traditional way.
+
+
+## Proposal
```

Comment:
The way we manage teams currently (using the old way, with static Jenkins and all that) is that we have 1 cluster and each team has their own Jenkins and environments so they don't step on each other's toes.  With this method, we do have sort of a "cluster team" in order to install cluster tools like prometheus, logging, etc.

Would this proposal support this sort of use case?  Seems like each team would have their own boot and then we would have a "cluster tools" boot that could install the shared cluster tools.

Reply:
I'll try improve the docs in the PR to show some samples of mult-team.

if teams are sharing the same cluster, its probably easiest to have a shared install of common things (lighthouse / tekton / nexus / prometheus / ingress / cert manager / vault et al), then have a few namespaces / CRDs per team.

We can store the team specific configuration in a team specific directory and let the folks in the teams be OWNERS of those folders; so that we can have 1 git repository to setup all the infrastructure + teams without having to have 1 git repo per team.

Reply:
Whaaaaaat!?! Please please show me the way!  This sounds amazing!  This quarter we are hoping to jump over to boot and all the new toys.  This would be a very good time to do such a thing.

I think our main hope or "requirement" was to have teams have their own environment repos so one teams deployment wouldn't disrupt another's.  We can also setup team quotas and other nifty things.

Reply:
@polothy I'll try get a demo going soon. All the heavy lifting is done now at least ;) Here's the current docs: https://github.com/jenkins-x/enhancements/tree/master/proposals/2/docs

Reply:
I'll re-review when I can.  This is looking very promising though, this is exactly what we need to help manage the cluster and teams.